api/src/main/java/org/openmrs/module/fhir/api/impl/DiagnosticReportServiceImpl.java

```diff
@@ -68,16 +93,31 @@ public void setDao(FHIRDAO dao) {
 	}
 
 	@Override
-	public DiagnosticReport getDiagnosticReport(String id) {
-		// Find Diagnostic Report (Encounter) in OpenMRS database
-		EncounterService encounterService = Context.getEncounterService();
-		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
-		// Get corresponding Handler
-		String handlerName = omrsDiagnosticReport.getEncounterType().getName();
+	public DiagnosticReport getDiagnosticReport(final String reportId) {
+		loadHandlerMap();
+		List<Order> orders = dao.getOrdersByAccessionNumber(reportId);
+		int orderId = 0;
+		if ((orders != null) && !orders.isEmpty()) {
+			orderId = orders.get(0).getOrderId();
+		}
+
+		if (orderId > 0) {
```

Comment:
Since `orderId` is just a number, how about if we just base this on the emptiness of the returned orders list?
That way, we do not mix this with the logic of what a valid order number should be. 😊